### PR TITLE
no more impostor comets

### PR DESCRIPTION
**Pull request by belisarius222**
Only treat an address as a comet if it's bigger than 64 bits.  Otherwise, a comet could impersonate a different kind of ship, which posed security problems.  This PR changes comet generation in Zuse, and comet validation in Ames.  I'm not aware of other places where this validation would be necessary, but it's possible others exist.

### CODE REVIEW

File: pkg/arvo/sys/vane/ames.hoon [hunk 1]
```diff
@@ -1056,6 +1056,9 @@
     ?>  ?=(@ content.packet)
     =+  ;;  [signature=@ signed=@]  (cue content.packet)
     =+  ;;  =open-packet            (cue signed)
+    ::  assert the comet can't pretend to be a moon or other address
+    ::
+    ?>  (gth 64 (met 0 sndr.packet))
```
Comment (ohAitch) [inline pkg/arvo/sys/vane/ames.hoon:1061]: Should this also just use `clan:title`? We probably don't want 129-bit "comets" either…

File: pkg/arvo/sys/zuse.hoon [hunk 2]
```diff
@@ -2173,6 +2173,74 @@
     ++  oath  @                                         ::  signature
     --  ::  pki
   --  ::  jael
+::                                                      ::::
+::::                      ++lenz                          ::  (1l) scry network
+  ::                                                    ::::
```
Comment (ohAitch) [inline pkg/arvo/sys/zuse.hoon:2178]: This core at minimum seems unrelated to the commit message

Comment (belisarius222) [reply pkg/arvo/sys/zuse.hoon:2178]: oops, cross-talk from another line of work -- will remove, apologies

File: pkg/arvo/sys/vane/ames.hoon (hunk 1 above)
Comment (belisarius222) [reply pkg/arvo/sys/vane/ames.hoon:1061]: unnecessary -- `sndr.packet` is read from a 128-bit slot in the packet, not from a jammed noun

**Comment by belisarius222** [comment]
ok, cleaned up the git snafu

**Comment by joemfb** [comment]
FWIW, i'd be really tempted to just write these as `?>  ?=(%pawn (clan:title some-ship))`.

**Comment by ohAitch** [comment]
Have the "is a bitwidth check" in the comments and "is a ship class check" in the code instead of the other way around :P

**Comment by belisarius222** [comment]
Ah, yeah I could add this check to `+clan:title` if it isn't implicitly there.  That would be cleaner.

**Comment by belisarius222** [comment]
redid it using `+clan:title`

**Comment by joemfb** [comment]
@pilfer-pandex, the same sanity check likely needs to be added to comet "mining" in king-haskell.

**Comment by ohAitch** [comment]
I mean, you're not going to mine 64 trailing zero bits on /accident/, this
seems mostly important to validate on the receiving end.

(Really if mining specific addresses were remotely feasible comets wouldn't
make sense as a concept, the worst you could hope for is hitting a random
moon which may or may not be spawned - but with a mature network this could
become, at all a viable grieving route via birthday attacks)

**Comment by belisarius222** [comment]
@ohAitch Mining a moon-length comet is extremely unlikely, but still invalid, so it's worth checking when trying to create a comet address -- otherwise it's wrong, and this is Kelvin-versioned kernel code that defines the spec.

@philipcmonk I've addressed review comments. PTAL

**Review by philipcmonk — APPROVED** [review]
LGTM
